final_application/inpainting_analysis_segmenter.py

```diff
@@ -0,0 +1,92 @@
+                 original_config_path: Optional[Path] = None, batch_size: Optional[int] = None,
+                 max_image_size: int = None, print_progress: bool = True, patch_overlap: int = 0,
+                 patch_overlap_factor: float = 0.0, show_confidence_in_segmentation: bool = False):
+        super().__init__(model_checkpoint=model_checkpoint, device=device, class_to_color_map=class_to_color_map,
+                         original_config_path=original_config_path, batch_size=batch_size,
+                         max_image_size=max_image_size, print_progress=print_progress,
+                         patch_overlap=patch_overlap, patch_overlap_factor=patch_overlap_factor,
+                         show_confidence_in_segmentation=show_confidence_in_segmentation)
+        self.patch_size = int(self.config['cutting_size'])
+
+    def crop_and_batch_patches(self, input_image: ImageClass, normalize: bool = True) -> Iterator[dict]:
+        if normalize:
+            transform_list = [
+                transforms.ToTensor(),
+                transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+            ]
+        else:
+            transform_list = [
+                transforms.ToTensor()
+            ]
+        transform_list = transforms.Compose(transform_list)
+
+        bboxes_for_patches = self.calculate_bboxes_for_patches(*input_image.size)
+        for i in range(0, len(bboxes_for_patches), self.batch_size):
+            batch_bboxes = bboxes_for_patches[i:i + self.batch_size] if self.batch_size > 1 else [bboxes_for_patches[i]]
+            batch_images = [input_image.crop(bbox) for bbox in batch_bboxes]
+            batch_images = [transform_list(image) for image in batch_images]
+            batch_images = torch.stack(batch_images, dim=0)
+            batch_images = batch_images.to(self.device)
+            yield {'images': batch_images, 'bboxes': batch_bboxes}
+
+    def predict_patches(self, patches: Iterator[dict]) -> [dict]:
+        predicted_patches = []
+        for batch in self.progress_bar(patches, desc="Predicting patches...", leave=False):
+            batch_interpolated = torch_functional.interpolate(batch['images'],
+                                                              (self.config['image_size'], self.config['image_size']))
+            with torch.no_grad():
+                prediction = self.network.predict(batch_interpolated)
+            prediction = torch_functional.interpolate(prediction, (self.patch_size, self.patch_size))
+            for i, bbox in enumerate(batch['bboxes']):
+                predicted_patches.append({
+                    "prediction": prediction[i],
+                    "bbox": bbox,
+                    "ground_truth": batch['images'][i]
+                })
+
+        return predicted_patches
+
+    def assemble_predictions(self, patches: List[dict], output_size: Tuple) -> torch.Tensor:
```

Comment:
The patches have to be unsqueezed with the LBAM output. It also skips the patch_overlap methods since there is none with the inpainting model.

Reply:
The unsqueezing part shouldn't be the problem. This could be done outside of the method. 

I'm wondering does the method work differently/not at all if there's no overlap? If not, then you could still reuse it.